Documentation/policy/lifecycle.rst

```diff
@@ -0,0 +1,88 @@
+.. _endpoint_lifecycle:
+.. _Endpoint Lifecycle:
+
+Endpoint Lifecycle
+==================
+
+This section specifies the lifecycle of Cilium endpoints.
+
+Every endpoint in Cilium is in one of the following states:
+
+* ``restoring``: The endpoint was started before Cilium started, and
+  Cilium is restoring its networking configuration.
+* ``waiting-for-identity``: Cilium is allocating a unique identity for
+  the endpoint.
+* ``waiting-to-regenerate``: The endpoint received an identity and is
+  waiting for its networking configuration to be (re)generated.
+* ``regenerating``: The endpoint's networking configuration is being
+  (re)generated. This includes programming BPF for that endpoint.
+* ``ready``: The endpoint's networking configuration has been
+  successfully (re)generated.
+* ``disconnecting``: The endpoint is being deleted.
+* ``disconnected``: The endpoint has been deleted.
+
+.. image:: ../images/cilium-endpoint-lifecycle.png
+   :scale: 50 %
+   :align: center
+
+The state of an endpoint can be queried using the ``cilium endpoint
+list`` and ``cilium endpoint get`` CLI commands.
+
+While an endpoint is running, it transitions between the
+``waiting-for-identity``, ``waiting-to-regenerate``, ``regenerating``,
+and ``ready`` states.  A transition into the ``waiting-for-identity``
+state indicates that the endpoint changed its identity.  A transition
+into the ``waiting-to-regenerate`` or ``regenerating`` state indicates
+that the policy to be enforced on the endpoint has changed because of
+an identity change or a policy change.
```

Comment:
I'd recommend mentioning configuration changes (i.e., Conntrack configuration, endpoint-specific configuration) that translate into changes to `#define` in the headerfiles for specific options also trigger regeneration.